CMakeLists.txt

```diff
@@ -26,13 +27,23 @@ if(RABIT_BUILD_TESTS)
   list(APPEND rabit_libs rabit_mock) # add to list to apply build settings, then remove
 endif()
 
-foreach(lib ${rabit_libs})
-  #include "./internal/utils.h"
-  target_include_directories(${lib} PUBLIC
-    "$<BUILD_INTERFACE:${CMAKE_CURRENT_LIST_DIR}/include>"
-    "$<BUILD_INTERFACE:${CMAKE_CURRENT_LIST_DIR}/dmlc-core/include>"
-    )
-endforeach()
+if(RABIT_BUILD_DMLC)
+  foreach(lib ${rabit_libs})
+    #include "./internal/utils.h"
+    target_include_directories(${lib} PUBLIC
+      "$<BUILD_INTERFACE:${CMAKE_CURRENT_LIST_DIR}/include>"
+      "$<BUILD_INTERFACE:${CMAKE_CURRENT_LIST_DIR}/dmlc-core/include>"
+      )
+  endforeach()
+else()
+  foreach(lib ${rabit_libs})
+    #include "./internal/utils.h"
+    target_include_directories(${lib} PUBLIC
+      "$<BUILD_INTERFACE:${CMAKE_CURRENT_LIST_DIR}/include>"
+      "$<BUILD_INTERFACE:${CMAKE_CURRENT_LIST_DIR}/../dmlc_core/include>"
```

Comment:
This is a really ugly hack. Don’t assume something outside of your project. This is what cmake is for. Set the correct include dir on the target using cmake.

Reply:
https://github.com/dmlc/rabit/pull/106/files